SonarSource/sonar-scanner-jenkins
# Update dependencies (incl. Jenkins 2.440.3) and fixed some deprecations

Update dependencies (incl. Jenkins 2.440.3) and fixed some deprecations and minor refactoring, after a short call with @okorach. The things I have done:
* bump plugin pom to latest version
* set Jenkins version to `2.440.3` according to [choosing-jenkins-baseline](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/) and 98% of the users which are on the latest plugin version have this according to the [stats](https://stats.jenkins.io/pluginversions/sonar.html).
* removed unused `joda-time`
* removed already configured `jacoco-maven-plugin` and use the version from the plugin pom
* replaced deprecated `Jenkins.getInstance()` with `Jenkins.get()`
* added some missing `@NonNull` annotations
* removed unused imports
* added Charsets to otherwise deprecated `Util.loadFile()`
* make use of diamond operator
* replaced `new String(Files.readAllBytes(p), StandardCharsets.UTF_8);` with newer `Files.readString(p);`
* fixed also some deprecations around `ACL:SYSTEM`


Please be aware that we are not actively looking for feature contributions. The truth is that it's extremely difficult for someone outside SonarSource to comply with our roadmap and expectations. Therefore, we typically only accept minor cosmetic changes and typo fixes. If you would like to see a new feature, please create a new thread in the forum ["Suggest new features"](https://community.sonarsource.com/c/suggestions/features).

With that in mind, if you would like to submit a code contribution, make sure that you adhere to the following guidelines and all tests are passing:

- [x] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/SONARJNKNS) ticket available, please make your commits and pull request start with the ticket ID (SONARJNKNS-XXXX)

We will try to give you feedback on your contribution as quickly as possible.

Thank You!
The SonarSource Team
